src/platform/packages/private/kbn-generate-csv/src/generate_csv_esql.test.ts

```diff
@@ -555,7 +556,80 @@ describe('CsvESQLGenerator', () => {
               }),
             ]),
             locale: 'en',
-            query: query.esql,
+            query: `${query.esql} | limit 500`,
+          },
+        },
+        {
+          strategy: 'esql',
+          transport: {
+            requestTimeout: '30s',
+          },
+          abortSignal: expect.any(AbortSignal),
+        }
+      );
+    });
+
+    it('adds maxRows limit to the query', async () => {
+      const query = {
+        esql: 'FROM custom-metrics | WHERE event.ingested >= ?_tstart AND event.ingested <= ?_tend',
+      };
+      const filters = [
+        {
+          meta: {},
+          query: {
+            range: {
+              'event.ingested': { format: 'strict_date_optional_time', gte: 'now-15m', lte: 'now' },
+            },
+          },
+        },
+      ];
```

Comment:
seems redundant for this test

Reply:
Updated in this commit, [44d1df7](https://github.com/elastic/kibana/pull/242603/commits/44d1df79eed7b085f598de03466a71c4a2bc1cc7)